pennylane/pulse/parametrized_evolution.py

```diff
@@ -505,6 +505,65 @@ def fun(y, t):
             mat = mat[-1]
         return qml.math.expand_matrix(mat, wires=self.wires, wire_order=wire_order)
 
+    def label(self, decimals=None, base_label=None, cache=None):
+        r"""A customizable string representation of the operator.
+
+        Args:
+            decimals=None (int): If ``None``, no parameters are included. Else,
+                specifies how to round the parameters.
+            base_label=None (str): overwrite the non-parameter component of the label
+            cache=None (dict): dictionary that carries information between label calls
+                in the same drawing
+
+        Returns:
+            str: label to use in drawings
+
+        **Example:**
+
+        >>> H = qml.PauliX(1) + qml.pulse.constant * qml.PauliY(0) + jnp.polyval * qml.PauliY(1)
+        >>> params = [0.2, [1, 2, 3]]
+        >>> op = qml.evolve(H)(params, t=2)
+
+        >>> op.label()
+        "ParametrizedEvolution"
+        >>> op.label(decimals=2)
+        "ParametrizedEvolution\n(p=[0.20,[1.00,2.00,3.00]], t=[0. 2.])"
+        >>> op.label(base_label="my_label")
+        "my_label"
+        >>> op.label(decimals=2, base_label="my_label")
+        "my_label\n(p=[0.20,[1.00,2.00,3.00]], t=[0. 2.])"
+        """
+        op_label = base_label or self.__class__.__name__
+
+        if self.num_params == 0:
+            return op_label
+
+        if decimals is None:
+            return op_label
+
+        params = self.parameters
+        arraylike_params = [p for p in params if qml.math.shape(p)]
+        has_cache = cache and isinstance(cache.get("matrices", None), list)
+
+        if arraylike_params and not has_cache:
```

Comment:
seems like you're not using `arraylike_params` anywhere else. just to make sure, you only want to support array-like parameters in labels _if_ a cache is given?
```suggestion
        has_cache = cache and isinstance(cache.get("matrices", None), list)

        if any(qml.math.ndim(p) for p in params) and not has_cache:
```

Reply:
That's correct! This is the same behaviour as on `Operation` for anything with a matrix as a parameter (e.g. `QubitUnitary`). If there are arrays/matrices and no cache, it only shows the base label.